be/src/runtime/runtime_state.cpp

```diff
@@ -305,6 +305,13 @@ void RuntimeState::get_unreported_errors(std::vector<std::string>* new_errors) {
     }
 }
 
+Status RuntimeState::query_status() {
+    auto st = _query_ctx->exec_status();
```

Comment:
why RETURN_IF_ERROR(_query_ctx->exec_status());

Reply:
When canceling pipeline fragment context or updating the status of pipeline fragment context, query context will be changed at the same time.
We use this status to try_close or close sink.